apps/api/src/person/person.service.ts

```diff
@@ -71,4 +75,20 @@ export class PersonService {
       Logger.warn(`Adding person to contacts list failed with code: ${error.code}`)
     }
   }
+
+  // Create/Update a marketing notifications consent for emails that are not registered
+  async upsertUnregisteredNotificationConsent(email: string, consent: boolean) {
+    await this.prisma.unregisteredNotificationConsent.upsert({
+      // Create new record
+      create: {
+        email,
+        consent,
+      },
+      // Update consent
+      update: { consent },
+      where: {
+        email,
+      },
+    })
+  }
```

Comment:
for the unregistered persons, I would expect this function to be in the NotificationsService because the PersonService is for registered users